client/state/data-layer/utils.js

```diff
@@ -21,22 +30,23 @@ const doBypassDataLayer = {
 export const bypassDataLayer = action => extendAction( action, doBypassDataLayer );
 
 /**
- * Deeply converts keys from the specified object to camelCase notation.
+ * Deeply converts keys of an object using provided function.
  *
  * @param {Object} obj object to convert
- * @returns {Object}   a new object with all keys converted
+ * @param  {Function} fn function to apply to each key of the object
+ * @returns {Object} a new object with all keys converted
  */
-export function convertToCamelCase( obj ) {
+export function convertKeysTo( obj, fn ) {
```

Comment:
**Minor**

`convertKeysBy` might be a bit more indicative that we'll be applying a supplied function to get our result. Lodash tends to have `…By` variants.

---

**Food for thought**

I'm *not* requesting this change, but I find it appealing and it's worth considering if you hadn't. This would break a bit with Lodash conventions. It's very appealing to curry and swap the order:

```js
export function makeKeyRewriter( fn ) {
  const keyRewriter = obj => {
     // snip…
  }
  return keyRewriter
}

export const convertToCamelCase = makeKeyRewriter( camelCase );
export const convertToSnakeCase = makeKeyRewriter( snakeCase );
```